go/chat_pub.go

```diff
@@ -0,0 +1,12 @@
+client, err := ably.NewRealtime(ably.WithKey("ABLY_API_KEY"))
+channel := client.Channels.Get('room:tesla-fans')
+pErr := channel.Presence.Enter(context.Background(), UserName + " entered the channel")
+if pErr != nil {
+	err := fmt.Errorf("error with enter presence on the channel %w", pErr)
+	fmt.Println(err)
+}
+err = channel.Publish(context.Background(), 'msg', "{'user': 'elon', 'message': 'Hello fans'}")
```

Comment:
Let's pass a `map[string]string` that will get serialised by the library rather than publishing a JSON string directly (since it will have the incorrect encoding), also also double quote `msg`:

```go
err = channel.Publish(context.Background(), "msg", map[string]string{"user": "elon", "message": "Hello fans"})
if err != nil {
	return fmt.Errorf("error publishing to the channel: %w", err)
}
```